src/engine/map/grid.ts

```diff
@@ -192,25 +197,22 @@ export class Grid {
     });
   }
 
-  connection(
+  /**
+   * Where neighbor is the location at fromCoordinates moved by direction, this
+   * will return the Track, if it exists, at the neighbor coordinates that connects
+   * back to fromCoordinates. If no track exists (e.g. because neighbor goes offboard,
+   * the neighbor tile is a city, or there is simply no track on that tile) undefined
+   * is returned.
+   */
+  getTrackConnection(
```

Comment:
Aren't there cases where we're looking for a City or an OwnedInterCityConnection? I'm not sure I see that being addressed in this refactor.

Reply:
Nevermind, I scanned the change, and I'm happy with the changes made.